src/cracker.c

```diff
@@ -1149,10 +1172,13 @@ static int process_key(char *key)
 			event_abort = event_pending = 1;
 	}
 
-	if (options.flags & FLG_MASK_STACKED)
-		mask_fix_state();
-	else
-	crk_fix_state();
+	if (fix_state_present && event_fix_state) {
+		if (options.flags & FLG_MASK_STACKED)
+			mask_fix_state();
+		else if (crk_fix_state)
+			crk_fix_state();
+		event_fix_state = 0;
+	}
```

Comment:
Same comments as I had on the previous occurrence of this snippet. And the code duplication is another reason to make this snippet shorter by moving some of the logic into `crk_init()`.